src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1455,6 +1456,43 @@ public ValidateSetAttribute(params string[] validValues)
 
             _validValues = validValues;
         }
+
+        /// <summary>
+        /// Initializes a new instance of the ValidateSetAttribute class
+        /// </summary>
+        /// <param name="type">list of valid values</param>
+        /// <exception cref="ArgumentNullException">for null arguments</exception>
+        /// <exception cref="ArgumentOutOfRangeException">for invalid arguments</exception>
+        public ValidateSetAttribute(Type type)
+        {
+            IValidateSetValuesGenerator validValuesGenerator;
+            if (typeof(IValidateSetValuesGenerator).IsAssignableFrom(type))
+            {
+                validValuesGenerator = (IValidateSetValuesGenerator)Activator.CreateInstance(type);
+            }
+            else
+            {
+                throw PSTraceSource.NewArgumentNullException("validValues");
```

Comment:
Cut and paste? I would use `ArgumentException` and please use the correct parameter name.